paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java

```diff
@@ -956,46 +986,46 @@ public boolean tryCommitOnce(
                                         // case
                                         !fileIO.exists(newSnapshotPath) && callable.call());
             } else {
+                if (useSerializableIsolation) {
+                    checkBeforeCommit(newSnapshotId);
+                }
                 success = callable.call();
+                if (useSerializableIsolation && success) {
+                    checkAfterCommit(newSnapshotId);
+                }
             }
-        } catch (Throwable e) {
+        } catch (CommitStateUnknownException e) {
             // exception when performing the atomic rename,
             // we cannot clean up because we can't determine the success
-            throw new RuntimeException(
-                    String.format(
-                            "Exception occurs when committing snapshot #%d (path %s) by user %s "
-                                    + "with identifier %s and kind %s. "
-                                    + "Cannot clean up because we can't determine the success.",
-                            newSnapshotId,
-                            newSnapshotPath,
-                            commitUser,
-                            identifier,
-                            commitKind.name()),
-                    e);
+            throw e;
+        } catch (DirtyCommitException e) {
+            // We need to clean up all the metadata information generated by this commit.
+            cleanUpTmpManifests(
+                    previousChangesListName,
+                    newChangesListName,
+                    changelogListName,
+                    newIndexManifest,
+                    oldMetas,
+                    newMetas,
+                    changelogMetas);
+            throw e;
+        } catch (Exception e) {
+            // If the commit succeeds, we ignore all exceptions thrown by the successful commit;
+            // otherwise, we throw an exception.
+            if (!success && useSerializableIsolation) {
+                throw new CommitFailedException(e.getMessage(), e);
+            }
```

Comment:
If the rename operation succeeds, we should ignore any exceptions thrown after that.